src/WordCloud/WordCloud.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useCallback, FC } from 'react';
+import cloud from 'd3-cloud';
+import { max } from 'd3-array';
+import { scaleLinear } from 'd3-scale';
+import { getColor, schemes } from '@/common/color';
+import { ChartShallowDataShape } from '@/common/data';
+import {
+  ChartProps,
+  ChartContainer,
+  ChartContextProps
+} from '@/common/containers';
+import { WordCloudLabel } from './WordCloudLabel';
+
+export interface WordCloudProps extends ChartProps {
+  /**
+   * Data to render in the word cloud.
+   * The 'key' represents the word and 'data' represents its frequency/weight.
+   */
+  data: ChartShallowDataShape[];
+
+  /**
+   * Font size range [min, max].
+   */
+  fontSizeRange?: [number, number];
+
+  /**
+   * Font family to use.
+   */
+  fontFamily?: string;
+
+  /**
+   * Padding between words.
+   */
+  padding?: number;
+
+  /**
+   * Rotation angles for words.
+   */
+  rotationAngles?: [number, number];
+
+  /**
+   * Number of possible rotations.
+   */
+  rotations?: number;
+
+  /**
+   * Custom color scheme for words.
+   */
+  colorScheme?: string[];
+
+  /**
+   * Event triggered when a word is clicked.
+   */
+  onLabelClick?: (event: React.MouseEvent, data: ChartShallowDataShape) => void;
+
+  /**
+   * Mouse enter handler.
+   */
+  onLabelMouseEnter?: (
+    event: React.PointerEvent,
+    data: ChartShallowDataShape
+  ) => void;
+
+  /**
+   * Mouse leave handler.
+   */
+  onLabelMouseLeave?: (
+    event: React.PointerEvent,
+    data: ChartShallowDataShape
+  ) => void;
+}
+
+export const WordCloud: FC<Partial<WordCloudProps>> = ({
```

Comment:
using `Partial` here doesn't seem right. wouldn't we want to just `Omit` when extending `ChartProps` the props that aren't going to be used?

```
Omit<ChartProps, 'id' | 'style' | 'center' | 'centerX' | 'centerY'>
```

technically wouldn't change anything since the `ChartProps` are all optional anyways but it should make it more clear to the user what props will actually affect something

Reply:
This is just how the others do it, we can address this in our API update.